sympy/parsing/tests/test_sympy_parser.py

```diff
@@ -280,16 +280,21 @@ def test_parse_function_issue_3539():
     assert parse_expr('f(x)') == f(x)
 
 def test_issue_24288():
-    inputs = {
-        "1 < 2": Lt(1, 2, evaluate=False),
-        "1 <= 2": Le(1, 2, evaluate=False),
-        "1 > 2": Gt(1, 2, evaluate=False),
-        "1 >= 2": Ge(1, 2, evaluate=False),
-        "1 != 2": Ne(1, 2, evaluate=False),
-        "1 == 2": Eq(1, 2, evaluate=False)
-    }
-    for text, result in inputs.items():
-        assert parse_expr(text, evaluate=False) == result
+    assert parse_expr("1 < 2", evaluate=False) == Lt(1, 2, evaluate=False)
+    assert parse_expr("1 <= 2", evaluate=False) == Le(1, 2, evaluate=False)
+    assert parse_expr("1 > 2", evaluate=False) == Gt(1, 2, evaluate=False)
+    assert parse_expr("1 >= 2", evaluate=False) == Ge(1, 2, evaluate=False)
+    assert parse_expr("1 != 2", evaluate=False) == Ne(1, 2, evaluate=False)
+    assert parse_expr("1 == 2", evaluate=False) == Eq(1, 2, evaluate=False)
+    assert parse_expr("1 < 2 < 3", evaluate=False) == And(Lt(1, 2, evaluate=False), Lt(2, 3, evaluate=False), evaluate=False)
+    assert parse_expr("1 <= 2 <= 3", evaluate=False) == And(Le(1, 2, evaluate=False), Le(2, 3, evaluate=False), evaluate=False)
+    assert parse_expr("1 < 2 <= 3 < 4", evaluate=False) == \
+        And(Lt(1, 2, evaluate=False), Le(2, 3, evaluate=False), Lt(3, 4, evaluate=False), evaluate=False)
+    # parse_expr with evaluate=True also falls back to Python comparison for them
+    assert parse_expr(r"1 in {1}", evaluate=False)
```

Comment:
I'm not sure if this should be allowed. It uses the Python set containment checking:
```python
In [1]: parse_expr('x in {y}')
Out[1]: False

In [2]: x in S({y})
...
TypeError: did not evaluate to a bool: Eq(y, x)

In [3]: Contains(x, {y})
Out[3]: x ∈ {y}
```
In principle `parse_expr` could use `Contains` but I don't know whether this should be treated as semantic containment checking or just that `parse_expr` represents what the Python code would do.

Reply:
Okay, I'd revert the commit and please revert from there.
I also think that supporting the sets is difficult.
For example, we also have to fix `parse_expr("{1} | {2}", evaluate=False)`, however, that requires other complicated implementation that distinguishes which operator to use (`Or, Union`) based on the pattern.

Reply:
I guess if `{1}` was parsed as `FiniteSet(1)` then the parser could depend on `__or__` to choose this but I would like to get away from having the parser evaluate arbitrary code in the long run.